src/styles/main.scss

```diff
@@ -0,0 +1,61 @@
+* {
+  font-family: Arial, sans-serif;
+}
+
+$height-block: 100px;
+$width-block: 100px;
+$amount-days-in-row: 7;
+$num-days: 31;
+
+body {
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+@for $i from 1 through $num-days {
+  .calendar {
+    display: flex;
+    gap: 1px;
+    flex-wrap: wrap;
+    max-width: ($width-block * $amount-days-in-row + $amount-days-in-row);
+
+    &__day {
+      box-sizing: border-box;
+      height: $height-block;
+      width: $width-block;
+      background-color: #eee;
+      border: 1px black solid;
+      text-align: center;
+      line-height: $height-block;
+      font-size: 30px;
+      transition: all 0.5s ease;
+
+      &:nth-child(#{$i})::before {
+        content: '#{$i}';
+      }
+
+      &:hover {
+        background-color: pink;
+        transform: translate(0, -20px);
+        cursor: pointer;
+      }
+    }
+  }
+}
```

Comment:
The `@for` loop is correctly used to generate styles for each day in the calendar. This approach is efficient for styling repetitive elements.